plugin/pkg/scheduler/scheduler.go

```diff
@@ -310,6 +382,11 @@ func (sched *Scheduler) scheduleOne() {
 
 	glog.V(3).Infof("Attempting to schedule pod: %v/%v", pod.Namespace, pod.Name)
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.VolumeTopologyBinding) {
+		// TODO: find a better way to do this
+		sched.config.VolumeBinder.InitTmpData(pod)
```

Comment:
@bsalamat this is another place I would greatly appreciate your feedback.

I want to be able to store some temporary data with the pod as it goes through the predicate->priorities->assumeVolume.  The way I have done this here is keep a temporary struct in the volume binder and clear it at the beginning of scheduleOne().

Another way I was considering was to keep my own pod cache with the temporary data.  But then I would also have to add a TTL to prevent memory leaks, which is more heavyweight than the current approach.

I need to store two pieces of information:
* Whether or not all of the pod's PVCs are bound at the beginning of scheduleOne().  This is to handle the case where PVCs can become bound in the middle of running predicates.  This is needed for volume predicate correctness because this PR is changing existing volume predicates to temporarily return true if the pod's PVCs are unbound, but they need to be re-evaluated once the volumes are actually bound.
* When the Pod's PVCs are unbound, cache the returned matching PVs from the VolumeBinding predicate per node.  That way, in the priority function (future), and in the assume volume function, we do not need to search again for the best PV matches.  Every time we search, we need to iterate through all the PVs.

Reply:
I think I can eliminate the first reason.  It was mainly to handle the volume zone predicate under the old design that applied to all PV binding.  But now that this new behavior is restricted, the volume zone predicate is only handling the PVs with the old behavior.  And the new behavior PVs will be handled with the volume bind predicate.  So there's only one predicate that we need to serialize bound state with.

But the 2nd reason for the PV matches cache is still good to have.  I could look into the TTL approach or use a pod informer to delete pods that no longer need scheduling.